python_modules/dagster/dagster/core/storage/runs/migration.py

```diff
@@ -51,3 +54,19 @@ def migrate_run_partition(storage, print_fn=None):
             continue
 
         storage.add_run_tags(run.run_id, run.tags)
+
+
+def migrate_run_start_end(storage, print_fn=None):
+    """
+    Utility method that updates the start and end times of historical runs using the completed event log.
+    """
+
+    if print_fn:
+        print_fn("Querying run and event log storage.")
+
+    for run in chunked_run_iterator(storage, print_fn):
```

Comment:
I think this will iterate over a chunked `get_runs` call, and then call `get_run_records` in a loop.  Might be more efficient to write a `chunked_run_records_iterator` instead, and avoid the extra record fetch.